### ISSUE TITLE
Provide filter to hook into purgeCacheByRelevantURLs results

### ISSUE THREAD

**Issue by piscis** [cloudflare/Cloudflare-WordPress#192]
we would like a way to purge custom urls based on the postID provided to `Wordpress\Hooks->purgeCacheByRevelantURLSs`.

I wonder if you would accept a PR that basically filters the result of `Hooks->getPostRelatedLinks($postId)` and give us a chance to add some more URLs to the list of URLs the plugin should purge.

**Example:**

      return apply_filters('cf_get_post_related_links', $listofurls, $postId);

**Comment by thellimist** [comment]
Sounds good. We try handle uniqueness of the filters with namespaces. Instead of using `cf_get_post_related_links` please use the naming we have used. 

Thanks

**Comment by piscis** [comment]
sorry I looked at the plugins source and did not find the appropriate namespace to use.
For example I found `apply_filters('http2_link_preload_src', $src);`  in `public static function http2LinkPreloadHeader($src)` but it seams to not use any namespace. 

Could you please tell me how I should declare the filter or point me to an example? 

Right now I declared the filter this way:

     return apply_filters('cf_get_post_related_links', $listofurls, $postId);

Thanks

**Comment by thellimist** [comment]
In [cloudflare.loader.php](https://github.com/cloudflare/Cloudflare-WordPress/blob/master/cloudflare.loader.php) you can see custom actions being added with namespaces.

I haven't tested but the code below should work.

```
$cloudflareHooks = new \CF\WordPress\Hooks();
add_action('cloudflare_get_post_related_links', array($cloudflareHooks, 'getPostRelatedLinks'), $listOfUrls, $postId);
```

Having `cloudflare` prefix rather than `cf` would avoid clashing with other Hooks which may use `cf` prefix. 

We can discuss better after the PR. In the description of the please let us know how you're using this Hook.

**Comment by jwineman** [comment]
@piscis,

To expand on this a little more:

The reason WordPress best practices suggest namespacing plugin specific functions is because those functions all live in the global namespace and that is the easiest way to avoid collisions.  Since  the functions that handle cache purging live inside [`CF\WordPress\Hooks`](https://github.com/cloudflare/Cloudflare-WordPress/blob/master/src/WordPress/Hooks.php) the function names don't need to be namespaced because they only exist inside the Hooks class scope and therefore can't be overridden by other plugins.

I'm not familiar with the filter but I think what you would want instead of
```
return apply_filters('cf_get_post_related_links', $listofurls, $postId);
```
would be something like this:
```
return apply_filters(array($cloudflareHooks, 'getPostRelatedLinks'), $listofurls, $postId);
```

**Comment by piscis** [comment]
Bei CF gibts dafür jetzt ein PR von mir: https://github.com/cloudflare/Cloudflare-WordPress/pull/195

**Comment by piscis** [comment]
can be closed due to the merge of #197

**Comment by jwineman** [comment]
Thanks!
